File: CustomizationEngine/lib/theme/engine_tokens.dart
import 'package:flutter/material.dart';

/// The few non-brand constants the engine's own resolvers need as
/// last-resort defaults, so the package depends on NEITHER consuming app's
/// `DesignConstants`. These are only fallbacks: every call site that has a
/// real value (icon size/weight/colour, a loaded brand colour) passes or
/// resolves it, and these never fire then.
///
/// Brand colours still resolve LIVE from the loaded customization via
/// `ThemeColor.token(...)`; the values here are just the const fallbacks
/// used when nothing is loaded (the white-label resilience property). They
/// mirror the CombatDen defaults in MobileApp's `design_constants.dart`.
class EngineTokens {
  EngineTokens._();

  /// Default icon edge length (matches `DesignConstants.iconSizeMd`).
  static const double iconSizeMd = 24;

  /// Material Symbol stroke weight (matches `DesignConstants.iconWeight`).
  static const double iconWeight = 300.0;

  /// Last-resort icon tint when no colour and no ambient `IconTheme` apply.
  /// The live default resolves `ThemeColor.token('text', ...)` against this.
  static const Color fallbackIconColor = Color(0xFFF4F3EE);
}

File: ThemeService/ThemeReact/src/lib/theme/engineTokens.ts
// Ports ../../ThemeFlutter/lib/theme/engine_tokens.dart.

import type { Rgba } from './color';
import { rgba } from './color';

/**
 * The few non-brand constants the runtime's OWN resolvers need as last-resort
 * defaults, so the library depends on neither consuming app's design constants.
 *
 * These are only fallbacks: every call site that has a real value (icon
 * size/colour, a loaded brand colour) passes or resolves it, and these never
 * fire then. Brand colours still resolve LIVE — the values here are what render
 * when nothing is loaded at all, which is the white-label resilience property.
 */
export const EngineTokens = Object.freeze({
  /** Default icon edge length, in px. */
  iconSizeMd: 24,

  /**
   * Last-resort icon tint when no colour prop and no loaded `text` token apply.
   * `#F4F3EE`, the CombatDen default.
   */
  fallbackIconColor: rgba(244, 243, 238) as Rgba,

  /**
   * The system stack every resolved font falls back through. Ported from
   * ThemeFlutter's implicit behaviour: `GoogleFonts.getFont` degrading to a
   * bare `TextStyle()` when a family will not resolve.
   */
  fallbackFontStack: 'system-ui, sans-serif',
});
